This will turn off sound and enable vibration of the dialpad.

Index: /usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/dialpad-assistant.js
@@ -634,7 +634,7 @@ var DialpadAssistant = Class.create({
 	numberClick: function(event, key){
 		
 		if (key) {
-			TelephonyCommands.sendDTMF(key, true);
+			this.controller.serviceRequest("palm://com.palm.vibrate", {method: 'vibrate', parameters: { 'period': 0,'duration': 100 }});
 			this.formatAndUpdateDialString(key,event);
 			if (this.dialString.length == 1 && this.speedDialString.length == 0) {
 				QDLogger.log( "DialpadAssistant#numberClick", "about to lookup speed dial");
@@ -929,7 +929,7 @@ var DialpadAssistant = Class.create({
 		if (this.isValidToneKey(this.pressedKey)
 			&& this.pressedKey !== Mojo.Char.backspace
 			&& (!this.appAssistant.lookupOnKeyPress || this.keyMapOpt.get(evt.keyIdentifier))) {
-			TelephonyCommands.sendDTMF(this.pressedKey, true);
+			this.controller.serviceRequest("palm://com.palm.vibrate", {method: 'vibrate', parameters: { 'period': 0,'duration': 100 }});
 		}
 		
 		Event.stop(evt);
